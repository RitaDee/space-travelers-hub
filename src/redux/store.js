import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './mission/MissionSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
