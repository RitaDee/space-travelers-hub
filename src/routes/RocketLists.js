import React from 'react';
import { useSelector } from 'react-redux';
import RocketUi from '../components/RocketUi';

const RocketLists = () => {
  const rockets = useSelector((store) => store.rockets?.list ?? []);
  return (
    <div>
      {rockets.map((rocket) => (
        <RocketUi
          key={rocket.rocket_id}
          id={rocket.rocket_id}
          name={rocket.rocket_name}
          desc={rocket.description}
          reserved={rocket.reserved}
          picture={rocket.flickr_images}
        />

      ))}
    </div>

  );
};

export default RocketLists;
