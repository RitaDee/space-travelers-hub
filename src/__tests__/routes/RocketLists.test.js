import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import RocketLists from '../../routes/RocketLists';

describe('RocketsLists', () => {
  it('renders correctly for reserved state', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <RocketLists />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
