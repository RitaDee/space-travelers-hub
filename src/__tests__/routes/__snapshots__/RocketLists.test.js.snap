// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsLists renders correctly for reserved state 1`] = `<div />`;
